build.gradle

```diff
@@ -1533,3 +1512,17 @@ task setup() {  //These must be strings so that we can do lazy resolution. Else
         dependsOn ':fmlonly:extractMapped'
     dependsOn ':forge:extractMapped'
 }
+
+//Also output FMLOnly when building a version.
+if (System.env.TEAMCITY_VERSION) {
+    //Only setup the CI environment if and only if the environment variables are set.
+    tasks.configureTeamCity {
+        doLast {
+            println "##teamcity[setParameter name='env.PUBLISHED_JAVA_FML_ARTIFACT_VERSION' value='${project(':fmlonly').version}']"
+        }
+    }
+}
+
+changelog {
+    fromTag "39.0"
```

Comment:
Should this possibly be declared in the `ext` block with other configurable version numbers?

Reply:
Likely yeah